checks/themename.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Check for prohibited words and theme names.
+ */
+
+class ThemeNameCheck implements themecheck {
+	protected $error = array();
+
+	function check( $php_files, $css_files, $other_files ) {
+
+		checkcount();
+		$ret = true;
+		global $themename;
+		global $data;
+
+		/* Remove the link added by get_theme_data_from_contents. */
+		$author = wp_strip_all_tags( $data['Author'] );
+
+		/* If the author is not the WordPress team, check for prohibited names. */
+		if ( 'the WordPress team' !== $author ) {
+
+			$blacklist = array( 'twentyten', 'twentyeleven', 'twentytwelve', 'twentythirteen', 'twentyfourteen', 'twentyfifteen',
+			'twentysixteen', 'twentyseventeen', 'twentyeighteen', 'twentynineteen', 'twentytwenty', 'twentytwentyone',
+			'twentytwentytwo', 'twentytwentythree', 'twentytwentyfour', 'twentytwentyfive', 'twentytwentysix',
+			'twentytwentyseven', 'twentytwentyeight', 'twentytwentynine', 'twentythirty'
+			);
+
+			foreach ( $blacklist as $key ) {
```

Comment:
```suggestion
			foreach ( $blacklist as $theme_name ) {
```

Technically it's array value, but it's always better to have a clear naming. So `$theme_name` fits better imo.